includes/discovery/route.inc.php

```diff
@@ -14,154 +14,330 @@
+                        unset($entry['mplsL3VpnVrfRteXCPointer']);
+                        unset($entry['mplsL3VpnVrfRteInetCidrMetric1']);
+                        unset($entry['mplsL3VpnVrfRteInetCidrMetric2']);
+                        unset($entry['mplsL3VpnVrfRteInetCidrMetric3']);
+                        unset($entry['mplsL3VpnVrfRteInetCidrMetric4']);
+                        unset($entry['mplsL3VpnVrfRteInetCidrMetric5']);
+                        unset($entry['mplsL3VpnVrfRteInetCidrAge']);
+                        unset($entry['mplsL3VpnVrfRteInetCidrProto']);
+                        unset($entry['mplsL3VpnVrfRteInetCidrType']);
+                        unset($entry['mplsL3VpnVrfRteInetCidrStatus']);
+                        unset($entry['mplsL3VpnVrfRteInetCidrIfIndex']);
+                        unset($entry['mplsL3VpnVrfRteInetCidrNextHopAS']);
+                        $current = $mixed[$vpnId][$inetCidrRouteDestType][$inetCidrRouteDest][$inetCidrRoutePfxLen][$inetCidrRoutePolicy][$inetCidrRouteNextHopType][$inetCidrRouteNextHop];
+                        if (isset($current) && isset($current['db']) && count($current['db']) > 0 && $delete_row[$current['db']['inetCidrRoute_id']] != 1) {
+                            //we already have a row in DB
+                            $entry['inetCidrRoute_id'] = $current['db']['inetCidrRoute_id'];
+                            $update_row[] = $entry;
+                        } else {
+                            d_echo(isset($current));
+                            d_echo(isset($current['db']));
+                            d_echo($current['db']);
+                            d_echo(count($current['db']));
+                            d_echo($delete_row[$current['db']['inetCidrRoute_id']]);
+                            $entry['created_at'] = array('NOW()');
+                            $create_row[] = $entry;
+                        }
                     }
                 }
-                if (!empty($changeRoute)) {
-                    dbUpdate($changeRoute, 'route', 'device_id = ? and ipRouteDest = ? and context_name = ?', array($device['device_id'], $ipRoute['ipRouteDest'], $device['context_name']));
-                }
-            } else {
-                $toInsert = array_merge($ipRoute, array('device_id' => $device['device_id'], 'context_name' => $device['context_name'], 'discoveredAt' => time()));
-                dbInsert($toInsert, 'route');
             }
         }
-        // unset($tableRoute);
     }
+}
+echo "\nProcessing: ";
 
-    unset($vrfs_lite_cisco);
+// We can now process the data into the DB
+foreach ($delete_row as $k => $v) {
+    if ($v > 0) {
+        dbDelete(
+            'inetCidrRoute',
+            '`inetCidrRoute_id` = ?',
+            array($k)
+        );
+        echo '-';
+        d_echo($delete_row_data[$k]);
+    }
 }
+
+foreach ($update_row as $upd_entry) {
+    dbUpdate(
```

Comment:
Using Eloquent would save a lot of code, up to you if you want to convert it as it will change a decent amount.

Reply:
I don't feel comfortable yet with Eloquent. Took me some time just to use the port relation without crashing with 100 lines of Exception :) I'll keep (testing) this code for now.

Reply:
no problem